shims/src/peripherals/input.rs

```diff
@@ -32,6 +41,34 @@ impl Input for InputShim {
             Err(_) => Err(ReadError),
         }
     }
+
+    fn register_interrupt_flag(&mut self, flag: &'a AtomicBool) {
+        self.flag = match self.flag {
+            None => Some(flag),
+            Some(_) => unreachable!(),
```

Comment:
So actually

Not sure if we want to allow setting the interrupt flag again

Reply:
See above